Add a way to force disable parser generation during build?

Currently the rizin build system checks for the python yaml module at configure time and if it finds it, regenerates the rizin command parser during the build.

I have an (admittedly niche) scenario (OpenBSD packaging with [dpb](http://man.openbsd.org/dpb)), where the yaml module may be present at configure time, but not at build time.

This is what we think caused the build failure:
```
/usr/local/bin/python3.9 ../rizin-v0.4.0/librz/core/cmd_descs/cmd_descs_generate.py --output-dir librz/core/cmd_descs --src-output-dir /usr/obj/ports/rizin-0.4.0/rizin-v0.4.0/librz/core/cmd_descs ../rizin-v0.4.0/librz/core/cmd_descs/cmd_analysis.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_block.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_cmp.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_debug.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_descs.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_egg.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_eval.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_flag.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_heap_glibc.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_history.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_info.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_interpret.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_meta.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_open.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_plugins.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_print.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_project.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_quit.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_remote.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_resize.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_seek.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_shell.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_system.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_tasks.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_type.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_write.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_yank.yaml ../rizin-v0.4.0/librz/core/cmd_descs/cmd_zign.yaml
Traceback (most recent call last):
  File "/usr/obj/ports/rizin-0.4.0/build-amd64/../rizin-v0.4.0/librz/core/cmd_descs/cmd_descs_generate.py", line 9, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
ninja: build stopped: subcommand failed.
```

For now we can work around this by registering the yaml module as a build dependency of rizin, but it would be nice if either:

 - the build system knew that regenerating the parser isn't necessary (timestamps?), or
 - there was a way to force off regeneration.

Thanks

--- a/librz/core/cmd_descs/meson.build
+++ b/librz/core/cmd_descs/meson.build
@@ -32,13 +32,17 @@ cmd_descs_yaml = files(
 cmd_descs_src_c = files('cmd_descs.c', 'cmd_descs.h')
 
 r = run_command(py3_exe, '-c', 'import yaml', check: false)
-if r.returncode() == 0
+cmd_option = get_option('regenerate_cmds')
+has_pyyaml = r.returncode() == 0
+if (cmd_option.auto() or cmd_option.enabled()) and has_pyyaml
   cmd_descs_ch = custom_target(
     'cmd_descs.[ch]',
     output: ['cmd_descs.c', 'cmd_descs.h'],
     input: cmd_descs_yaml,
     command: [py3_exe, cmd_descs_generate_py, '--output-dir', '@OUTDIR@', '--src-output-dir', meson.current_source_dir(), '@INPUT@']
   )
+elif cmd_option.enabled() and not has_pyyaml
+  error('PyYAML python module was not found, but it is required to regenerate cmd_descs.[ch]')
 else
   warning('PyYAML python module was not found, using cmd_descs.c/cmd_descs.h from source directory. Install PyYAML (either from your package manager or through pip) if you need to modify cmd_descs files.')
   cmd_descs_ch = cmd_descs_src_c

--- a/meson_options.txt
+++ b/meson_options.txt
@@ -39,3 +39,4 @@ option('debugger', type: 'boolean', value: true)
 
 option('enable_tests', type: 'boolean', value: true, description: 'Build unit tests in test/unit')
 option('enable_rz_test', type: 'boolean', value: true, description: 'Build rz-test executable for regression testing')
+option('regenerate_cmds', type: 'feature', value: 'auto', description: 'Regenerate the cmd_descs.[ch] files (requires PyYAML)')

